### ISSUE TITLE
HTML encoded "&" in URI queries

### ISSUE THREAD

**Issue by dev-101** [osclass/Osclass#1925]
This is huge, a lot of URIs in core files have html encoding in URL query parameter separators which is not standard.

e.g.
?page=users&action=status_alerts`&amp;`alert_id[]=...

& above is a query separator, not an HTML character.

**Comment by dev-101** [comment]
closed

**Comment by dev-101** [comment]
/after comment/

There is, however, a lack of consistency as it seems (all & should be encoded in such cases, not only half of the strings as it is now).

**Comment by conejoninja** [comment]
If this is related to _server_params branch and fix, then , it's not "lack" of consistency. As some params are forced in to the url (manually or by the code) and the rest of the URL came from the $_SERVER variables that were being sanitized. If this is related to #1928 we're looking into it.

**Comment by dev-101** [comment]
Hi Conejo,
first, this is not a big deal, it's a W3C c*...

second, no, not related to latest development, this is old news.

third, just perform a massive search in admin and core files for e.g. "?page=" and you will see. The above URL given in opening of this issue is just one example, first & character is not html encoded, the second one is. That is inconsistent.

fourth, that's only for admin side, so doesn't matter. If you wish to invest time, it would be better to fix front side (I have tried replacing manually each file that matters) but ended up with item url's not working in the end:

e.g. 
`http://website.com/index.php?page=item&amp;id=151`

With all changes in key files + hDefines, Osclass internally generated `&amp;' instead of '&'  conversion inside item URL, so all I got was 404. And for w3c...

my advice: forget it, leave it. I actually removed all `&amp;' to keep code cleaner, don't care.

Hope I explained :)
Regards
